### DISCUSSION TITLE
arvalid, rready 在最後兩筆測資時又 toggle

### DISCUSSION THREAD

**Discussion by chengyu-123** [bol-edu/HLS-SOC-Discussions#107] [Q&A]
我想請問有人也有遇到在送最後兩筆 ss_tdata 時， arvalid, rready 又toggle的問題嗎，會造成錯誤的 handshake( tap_A 會送錯)，謝謝~~
<img width="515" alt="image" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/69706789/a4e3420c-b499-4c66-8b92-c595b7306ffa">
<img width="345" alt="image" src="https://github.com/bol-edu/HLS-SOC-Discussions/assets/69706789/22620770-bf84-4bf4-bf1a-09c46487d1e4">

**Comment by ZheChen-Bill** [comment]
在你的系統正在運算狀態時，我們依然應該可以透過axilite給予arvalid、rready去讀取現在是不是在idle狀態的訊號，因此我們有在範例的testbench中做這件事情，會需要調整你的設計去符合這個功能。
